website/docs/advanced-concepts/granular-access-control/reference/permissions.mdx

```diff
@@ -0,0 +1,717 @@
+          <td>NA</td>
+          <td>Allows running queries within the given workspace.</td>
+        </tr>
+        <tr>
+          <td><b>Application</b></td>
+          <td>NA</td>
+          <td>Allows running queries within the given application.</td>
+        </tr>
+        <tr>
+          <td><b>Page</b></td>
+          <td>NA</td>
+          <td>Allows running queries within the given application.</td>
+        </tr>
+      </tbody>
+    </table>
+  </TabItem>
+  <TabItem label="Datasources & Environments" value="execute-datasources-and-environments">
+    <table className="permissions-table" style={{ width: '100%', display: 'table', marginLeft: '2px' }}>
+      <thead>
+        <tr>
+          <th>Resource</th>
+          <th>Execute also grants below permissions</th>
+          <th>Permission Impact</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td><b>Datasources</b></td>
+          <td>NA</td>
+          <td>Allows running queries in all datasources within the workspace.</td>
+        </tr>
+        <tr>
+          <td>&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;Application &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Datasource</td>
+          <td>NA</td>
+          <td>Allows running queries in the given datasource.</td>
+        </tr>
+        <tr>
+          <td><b>Environments</b></td>
+          <td>NA</td>
+          <td>Allows running queries in all environments within the workspace.</td>
+        </tr>
+        <tr>
+          <td>&nbsp;&nbsp;&nbsp;&nbsp; &nbsp;Environment</td>
+          <td>NA</td>
+          <td>Allows running queries in the given environment within the workspace.</td>
+        </tr>
+      </tbody>
+    </table>
+  </TabItem>
+</Tabs>
+
+## Make Public permission
+
+The **Make Public** permission permits users to change the access level of resources, making them available to the public. When the **Make Public** permission is assigned to a workspace or an application, it, along with the automatically assigned permissions, are inherited by the child resources. Once granted, users can enable the **Make application public** option on the _Invite Users_ modal to make an application publicly available. This option allows external users to access the app without the need to log in to Appsmith.
+
+<Tabs queryString="gac-resource">
+  <TabItem label="Application Resources" value="make-public-application-resources">
+    <table className="permissions-table" style={{ width: '100%', display: 'table', marginLeft: '2px' }}>
+      <thead>
+        <tr>
+          <th>Resource</th>
+          <th>Make Public also grants below permissions</th>
+          <th>Permission Impact</th>
+        </tr>
+      </thead>
+      <tbody>
+        <tr>
+          <td><b>Workspace</b></td>
+          <td>View, Execute</td>
+          <td>Allows making all applications publicly accessible within the given workspace, and allows viewing and running queries to view the application when applications are publicly accessible.</td>
```

Comment:
Make public permission will allow you to make an application public, let's only focus on that. The permission itself doesn't give access to other things you're talking about above.

Reply:
It gives view and execute permission so that the publicly available application is functional

Reply:
Sorry, no. This role doesn't. Once an app is made public, there's another permission set that is generated on the server that gives permissions to access public apps.

@trishaanand  to confirm.

Reply:
This is from the configuration roles page. When you give the make public permission - it does grant the view and execute permissions. Here's a [loom recording](https://www.loom.com/share/75f0f79d7c4f4e9fbbbd34bc82bfecd3?sid=c8b022a2-0280-4e58-8f09-8e1479395bb6) for the same.

Reply:
Yes but, that has nothing to with how end anonymous users get access to that public app. That is handled by another anonymous permissions.

Reply:
If these permissions are not granted wouldn't it impact the end user experience also. As the end user will not have view and execute permissions, and app will not be viewed and day will not be fetched as queries don't have execute permission. Why this anonymous permission is not available in the configuration?

Reply:
@jnikhila : If you turn on Make Public permission, then whoever is assigned this role would be given the ability to toggle the application between public and private. This itself doesnt make an application public. That CTA to make an application public/private is present within the application share modal.

Reply:
That's right. That part is added to the description of the permission. Could you please also confirm about the permission impact.